apps/web/app/(ee)/api/cron/calculate-program-similarities/calculate-category-similarity.ts

```diff
@@ -0,0 +1,43 @@
+import { prisma } from "@dub/prisma";
+
+// Calculate category similarity using Jaccard similarity
+export async function calculateCategorySimilarity(
+  program1Id: string,
+  program2Id: string,
+): Promise<number> {
+  const [categories1, categories2] = await Promise.all([
+    prisma.programCategory.findMany({
+      where: {
+        programId: program1Id,
+      },
+      select: {
+        category: true,
+      },
+    }),
+
+    prisma.programCategory.findMany({
+      where: {
+        programId: program2Id,
+      },
+      select: {
+        category: true,
+      },
+    }),
+  ]);
+
+  const categories1Set = new Set(categories1.map(({ category }) => category));
+  const categories2Set = new Set(categories2.map(({ category }) => category));
+
+  const sharedCount = [...categories1Set].filter((c) =>
+    categories2Set.has(c),
+  ).length;
+
+  const totalUniqueCount =
+    categories1Set.size + categories2Set.size - sharedCount;
+
+  if (totalUniqueCount === 0) {
+    return 0;
+  }
```

Comment:
I think this would return a 0 if both sets of categories match exactly, right?

Should it be something like this instead?

```
if (sharedCount === 0) return 0;
if (totalUniqueCount === 0) return 1;